common/src/main/java/com/wynntils/wc/objects/items/IdentificationContainer.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wc.objects.items;
+
+import com.wynntils.core.Reference;
+import com.wynntils.utils.StringUtils;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This code is largely based off info provided in
+ * https://forums.wynncraft.com/threads/how-identifications-are-calculated.128923/
+ */
+public class IdentificationContainer {
+
+    private static final Map<String, IdentificationModifier> typeMap = new HashMap<>();
+
+    protected IdentificationModifier type;
+    private final int baseValue;
+    protected boolean isFixed;
+
+    private transient int min, max;
+
+    public IdentificationContainer(IdentificationModifier type, int baseValue, boolean isFixed) {
+        this.type = type;
+        this.baseValue = baseValue;
+        this.isFixed = isFixed;
+        calculateMinMax();
+    }
+
+    public void calculateMinMax() {
+        if (isFixed || (-1 <= baseValue && baseValue <= 1)) {
+            min = max = baseValue;
+            return;
+        }
+
+        min = (int) Math.round(baseValue * (baseValue < 0 ? 1.3 : 0.3));
+        max = (int) Math.round(baseValue * (baseValue < 0 ? 0.7 : 1.3));
+    }
+
+    public void registerIdType(String name) {
+        if (typeMap.containsKey(name)) return;
+        typeMap.put(name, type);
+    }
+
+    public IdentificationModifier getType() {
+        return type;
+    }
+
+    public int getMax() {
+        return max;
+    }
+
+    public int getMin() {
+        return min;
+    }
+
+    public int getBaseValue() {
+        return baseValue;
+    }
+
+    public boolean isFixed() {
+        return isFixed;
+    }
+
+    public boolean hasConstantValue() {
+        return isFixed || min == max;
+    }
+
+    public static String getAsLongName(String shortName) {
+        if (shortName.startsWith("raw")) {
+            shortName = shortName.substring(3);
+            shortName = Character.toLowerCase(shortName.charAt(0)) + shortName.substring(1);
```

Comment:
Maybe make a `StringUtils` method for this too?

Reply:
I don't currently see the need, but if this code will be needed elsewhere, we can move it.